include/nfc/ndef/ch_msg.h

```diff
@@ -0,0 +1,155 @@
+/*
+ * Copyright (c) 2020 Nordic Semiconductor ASA
+ *
+ * SPDX-License-Identifier: LicenseRef-BSD-5-Clause-Nordic
+ */
+
+#ifndef NFC_NDEF_CH_MSG_H_
+#define NFC_NDEF_CH_MSG_H_
+
+/**@file
+ *
+ * @defgroup nfc_ndef_ch_msg NFC Connection Hanover pairing messages
+ * @{
+ * @ingroup nfc_ndef_messages
+ *
+ * @brief Generation of The Connection Handover pairing NDEF messages.
+ *
+ */
+
+#include <stddef.h>
+#include <zephyr/types.h>
+#include <nfc/ndef/ch.h>
+#include <nfc/ndef/le_oob_rec.h>
+
+#ifdef _cplusplus
+extern "C" {
+#endif
+
+/**
+ * @brief Encode an NFC NDEF LE OOB message.
+ *
+ * This function encodes an NFC NDEF message into a buffer.
+ *
+ * @param[in] oob Pointer to the LE OOB Record payload descriptor.
+ * @param[out] buf Pointer to the buffer for the message.
+ * @param[in, out] len Size of the available memory for the message as input.
+ *                     Size of the generated message as output.
+ *
+ * @retval 0 If the operation was successful.
+ *           Otherwise, a (negative) error code is returned.
+ */
+int nfc_ndef_ch_msg_le_oob_encode(struct nfc_ndef_le_oob_rec_payload_desc *oob,
+				  u8_t *buf, size_t *len);
+
+/**
+ * @brief Encode an NFC NDEF Handover Select message.
+ *
+ * This function encodes an NFC NDEF message into a buffer. It
+ * create a nested NDEF message with the Alternative Carrier Records.
+ * Next the Records with the Carrier data are encoded into the main
+ * Connection Hanover Select message. Number of the Alternative
+ * Carrier Records must be equal to the number of the Carrier Records.
+ *
+ * @param[in] ac Pointer to the Alternative Carrier Records.
+ * @param[in] carrier Pointer to the Carrier Records which identifies
```

Comment:
Can this be clarified? According to the code, this argument can be a pointer to element 0 of an array of Carrier Record descriptors. This applies to the series of functions _Hs, Hm, Hr, Hi_

Reply:
I added check for NULL pointer and count according to CH specification records without Alternative Carrier and Carrier record does not make a lot of sense.